comp/dogstatsd/server/rc.go

```diff
@@ -0,0 +1,92 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package server implements a component to run the dogstatsd server
+package server
+
+import (
+	"encoding/json"
+	"maps"
+	"slices"
+
+	"github.com/DataDog/datadog-agent/pkg/remoteconfig/state"
+)
+
+type statsdBlocklistUpdate struct {
+	BlockedMetrics blockedMetrics `json:"blocked_metrics"`
+}
+
+type blockedMetrics struct {
+	ByName byName `json:"by_name"`
+}
+
+type byName struct {
+	Metrics []metricEntry `json:"values"`
+}
+
+type metricEntry struct {
+	Name string `json:"metric_name"`
+}
+
+func (s *server) onBlocklistUpdateCallback(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debug("onBlocklistUpdateCallback received updates:", len(updates))
+
+	// special case: we received a response from RC, but RC didn't have any
+	// configuration for this agent, let's restore the local config and return
+	if len(updates) == 0 {
+		s.restoreBlocklistFromLocalConfig()
+		return
+	}
```

Comment:
You could potentially also skip this and just let the logic at the bottom of the method handle it, since `metricNames` will be empty if there were no updates.

Reply:
If you don't mind, I'll prefer to stick with an early return here.